internal/indexer/indexer_buffer.go

```diff
@@ -126,6 +148,18 @@ func (b *IndexerBuffer) PushStateChanges(stateChanges []types.StateChange) {
 	defer b.mu.Unlock()
 
 	b.stateChanges = append(b.stateChanges, stateChanges...)
+
+	for _, stateChange := range stateChanges {
```

Comment:
Good idea - using state changes to push participants since we cant extract everything.